### PR TITLE
Fixes related to section syntax

### PR DESCRIPTION
**Pull request by etienne-lms**
minor fixes in `architecture/trusted_applications.rst` and `build/toolchains.rst`.
* Replace `===` underlines with `---` underlines for subsection parts.
* Remove section/subsection numbering.
* Move local references to end of .rst file.

### CODE REVIEW

**Comment by etienne-lms** [comment]
@jbech-linaro : consider this P-R?

File: architecture/globalplatform_api.rst [hunk 1]
```diff
@@ -132,7 +135,8 @@ to operate with the data cache:
     TEE_Result TEE_CacheInvalidate(char *buf, size_t len);
 
 These functions are available to any Trusted Application defined with the flag
-``TA_FLAG_CACHE_MAINTENANCE`` sets on. When not set, each function returns the
+``TA_FLAG_CACHE_MAINTENANCE`` sets on, see :ref:`ta_property_cache_maintenance`.
+When not set, each function returns the
```
Comment (jbech-linaro) [inline architecture/globalplatform_api.rst]: Can you wrap this and the rest of the lines in this section on 80 character?

Comment (etienne-lms) [reply architecture/globalplatform_api.rst]: ok

**Comment by etienne-lms** [comment]
description wrapped.
tag applied.

**Pull request merged**